minichef/contracts/Strategy.sol

```diff
@@ -0,0 +1,410 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity 0.6.12;
+pragma experimental ABIEncoderV2;
+
+import "./interfaces/IMiniChefV2.sol";
+import "./interfaces/IUniswapV2Router02.sol";
+import "./interfaces/IUniswapV2Factory.sol";
+
+import "@tesrvaults/contracts/BaseStrategy.sol";
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts/math/Math.sol";
+
+
+contract Strategy is BaseStrategy {
+    using SafeERC20 for IERC20;
+    using Address for address;
+    using SafeMath for uint256;
+
+    address public targetToken;
+    // address public poolToken = IERC20(0x7479e1bc2f2473f9e78c89b4210eb6d55d33b645);
+
+    uint256 public pid;
+    uint256 public optimal = 2;
+
+    string internal stratName;
+    bool internal isOriginal = true;
+
+
+    IERC20 internal constant wavax = IERC20(0xB31f66AA3C1e785363F0875A1B74E27b85FD66c7);
+    IERC20 internal constant usdt = IERC20(0xc7198437980c041c805A1EDcbA50c1Ce5db95118);
+    IERC20 internal constant usdc = IERC20(0xB97EF9Ef8734C71904D8002F8b6Bc66Dd9c48a6E);
+    IERC20 internal constant dai = IERC20(0xd586E7F844cEa2F87f50152665BCbc2C279D8d70);
+    IERC20 internal poolToken = IERC20(0xCA87BF3ec55372D9540437d7a86a7750B42C02f4);
+    IERC20 public constant emissionToken = IERC20(0x1f1E7c893855525b303f99bDF5c3c05Be09ca251);
+
+    IUniswapV2Router02 public constant router = IUniswapV2Router02(0x60aE616a2155Ee3d9A68541Ba4544862310933d4); // traderjoe router
+    IMiniChefV2 public constant chef = IMiniChefV2(0x3a01521F8E7F012eB37eAAf1cb9490a5d9e18249);
+
+
+    bool internal forceHarvestTriggerOnce; // only set this to true externally when we want to trigger our keepers to harvest for us
+    uint256 public minHarvestCredit; // if we hit this amount of credit, harvest the strategy
+
+    event Cloned(address indexed clone);
+
+
+    constructor(
+        address _vault,
+        uint256 _pid,
+        string memory _name
+    ) public BaseStrategy(_vault) {
+        _initializeStrat(_pid, _name);
+    }
+
+    function initialize(
+        address _vault,
+        address _strategist,
+        address _rewards,
+        address _keeper,
+        uint256 _pid,
+        string memory _name
+    ) external {
+        _initialize(_vault, _strategist, _rewards, _keeper);
+        _initializeStrat(_pid, _name);
+    }
+
+    
+    function _initializeStrat(
+        uint256 _pid,
+        string memory _name
+    ) internal {
+        // initialize variables
+        maxReportDelay = 43200; // 1/2 day in seconds, if we hit this then harvestTrigger = True
+        healthCheck = address(0xf13Cd6887C62B5beC145e30c38c4938c5E627fe0); // Fantom common health check
```

Comment:
I think this health check does not exists on AVAX. The address is a valid smart contract on Fantom.

Reply:
updated with the exact one